jib-core/src/main/java/com/google/cloud/tools/jib/tar/TarStreamBuilder.java

```diff
@@ -56,45 +46,35 @@ private void writeEntriesAsTarArchive(OutputStream tarByteStream) throws IOExcep
         new TarArchiveOutputStream(tarByteStream, StandardCharsets.UTF_8.name())) {
       // Enables PAX extended headers to support long file names.
       tarArchiveOutputStream.setLongFileMode(TarArchiveOutputStream.LONGFILE_POSIX);
-      for (Map.Entry<TarArchiveEntry, TarArchiveOutputStreamConsumer> entry :
-          archiveMap.entrySet()) {
+      for (Map.Entry<TarArchiveEntry, Blob> entry : archiveMap.entrySet()) {
         tarArchiveOutputStream.putArchiveEntry(entry.getKey());
-        entry.getValue().accept(tarArchiveOutputStream);
+        entry.getValue().writeTo(tarArchiveOutputStream);
         tarArchiveOutputStream.closeArchiveEntry();
       }
     }
   }
 
   /**
-   * Adds an entry to the archive.
+   * Adds a {@link TarArchiveEntry} to the archive.
    *
-   * @param entry the entry to add.
+   * @param entry the {@link TarArchiveEntry}
    */
-  public void addEntry(TarArchiveEntry entry) {
+  public void addTarArchiveEntry(TarArchiveEntry entry) {
     archiveMap.put(
-        entry,
-        tarArchiveOutputStream -> {
-          // Note that this will skip files that don't exist.
-          if (entry.isFile()) {
-            try (InputStream contentStream =
-                new BufferedInputStream(Files.newInputStream(entry.getFile().toPath()))) {
-              ByteStreams.copy(contentStream, tarArchiveOutputStream);
-            }
-          }
-        });
+        entry, entry.isFile() ? Blobs.from(entry.getFile().toPath()) : Blobs.from(ignored -> {}));
   }
 
   /**
-   * Adds a blob to the archive.
+   * Adds a blob to the archive. Note that this should be used with raw bytes and not file contents;
+   * for adding files to the archive, use {@code TarStreamBuilder#addEntry(TarArchiveEntry entry)}.
    *
-   * @param contents the blob contents to add to the tarball.
-   * @param name the name of the entry (i.e. filename).
+   * @param contents the bytes to add to the tarball
+   * @param name the UTF-8 encoded name of the entry (i.e. filename)
```

Comment:
The "UTF-8 encoded" doesn't sound right. I see `TarArchiveEntry` just accepts a `String`, and we are just good. Basically, a `String` in memory in Java can be thought of as a conceptual thing irrelevant of encoding. The encoding matters only when you must write it down in bytes.

Reply:
Oh ok, makes sense.